plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -160,6 +167,32 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
         return new ConnectorTableMetadata(schemaTableName, columns);
     }
 
+    @Override
+    public Optional<SystemTable> getSystemTable(ConnectorSession session, SchemaTableName tableName)
+    {
+        if (SystemTableHandler.BIG_QUERY_VIEWS.matches(tableName)) {
+            return getBigQueryViewsSystemTable(tableName, SystemTableHandler.BIG_QUERY_VIEWS.getSourceTableName(tableName));
+        }
+        return Optional.empty();
+    }
+
+    private Optional<SystemTable> getBigQueryViewsSystemTable(SchemaTableName viewName, SchemaTableName sourceTableName)
```

Comment:
Is the name `viewName` accurate? That represents something like `customer_view$big_query_views`, no? In that case, it would be the "view definition table name", not the actual view name.